linode/databasemysqlv2/framework_resource.go

```diff
@@ -344,6 +346,54 @@ func (r *Resource) Update(
+		!helper.FrameworkValuesShallowEqual(state.EngineConfigMySQLInnoDBFTServerStopwordTable, plan.EngineConfigMySQLInnoDBFTServerStopwordTable),
+		!helper.FrameworkValuesShallowEqual(state.EngineConfigMySQLInnoDBLockWaitTimeout, plan.EngineConfigMySQLInnoDBLockWaitTimeout),
+		!helper.FrameworkValuesShallowEqual(state.EngineConfigMySQLInnoDBLogBufferSize, plan.EngineConfigMySQLInnoDBLogBufferSize),
+		!helper.FrameworkValuesShallowEqual(state.EngineConfigMySQLInnoDBOnlineAlterLogMaxSize, plan.EngineConfigMySQLInnoDBOnlineAlterLogMaxSize),
+		!helper.FrameworkValuesShallowEqual(state.EngineConfigMySQLInnoDBReadIOThreads, plan.EngineConfigMySQLInnoDBReadIOThreads),
+		!helper.FrameworkValuesShallowEqual(state.EngineConfigMySQLInnoDBRollbackOnTimeout, plan.EngineConfigMySQLInnoDBRollbackOnTimeout),
+		!helper.FrameworkValuesShallowEqual(state.EngineConfigMySQLInnoDBThreadConcurrency, plan.EngineConfigMySQLInnoDBThreadConcurrency),
+		!helper.FrameworkValuesShallowEqual(state.EngineConfigMySQLInnoDBWriteIOThreads, plan.EngineConfigMySQLInnoDBWriteIOThreads),
+		!helper.FrameworkValuesShallowEqual(state.EngineConfigMySQLInteractiveTimeout, plan.EngineConfigMySQLInteractiveTimeout),
+		!helper.FrameworkValuesShallowEqual(state.EngineConfigMySQLInternalTmpMemStorageEngine, plan.EngineConfigMySQLInternalTmpMemStorageEngine),
+		!helper.FrameworkValuesShallowEqual(state.EngineConfigMySQLMaxAllowedPacket, plan.EngineConfigMySQLMaxAllowedPacket),
+		!helper.FrameworkValuesShallowEqual(state.EngineConfigMySQLMaxHeapTableSize, plan.EngineConfigMySQLMaxHeapTableSize),
+		!helper.FrameworkValuesShallowEqual(state.EngineConfigMySQLNetBufferLength, plan.EngineConfigMySQLNetBufferLength),
+		!helper.FrameworkValuesShallowEqual(state.EngineConfigMySQLNetReadTimeout, plan.EngineConfigMySQLNetReadTimeout),
+		!helper.FrameworkValuesShallowEqual(state.EngineConfigMySQLNetWriteTimeout, plan.EngineConfigMySQLNetWriteTimeout),
+		!helper.FrameworkValuesShallowEqual(state.EngineConfigMySQLSortBufferSize, plan.EngineConfigMySQLSortBufferSize),
+		!helper.FrameworkValuesShallowEqual(state.EngineConfigMySQLSQLMode, plan.EngineConfigMySQLSQLMode),
+		!helper.FrameworkValuesShallowEqual(state.EngineConfigMySQLSQLRequirePrimaryKey, plan.EngineConfigMySQLSQLRequirePrimaryKey),
+		!helper.FrameworkValuesShallowEqual(state.EngineConfigMySQLTmpTableSize, plan.EngineConfigMySQLTmpTableSize),
+		!helper.FrameworkValuesShallowEqual(state.EngineConfigMySQLWaitTimeout, plan.EngineConfigMySQLWaitTimeout),
+	}
+	for _, changed := range engineConfigFields {
+		if changed {
+			shouldUpdate = true
+
+			engineConfig := plan.GetEngineConfig(resp.Diagnostics)
+			if resp.Diagnostics.HasError() {
+				return
+			}
+
+			updateOpts.EngineConfig = engineConfig
+			if resp.Diagnostics.HasError() {
+				return
+			}
+			break // already handled all updates in one call to GetEngineConfig()
+		}
+	}
```

Comment:
nitpick: We can tidy this up a bit using `slices.Contains`
```suggestion
	if slices.Contains(engineConfigFields, true) {
		shouldUpdate = true

		engineConfig := plan.GetEngineConfig(resp.Diagnostics)
		if resp.Diagnostics.HasError() {
			return
		}

		updateOpts.EngineConfig = engineConfig
		if resp.Diagnostics.HasError() {
			return
		}
	}
```

Reply:
Fixed